var/spack/repos/builtin/packages/rocm-openmp-extras/package.py

```diff
@@ -0,0 +1,337 @@
+                '{0}/amdgcn/bitcode'.format(format(devlibs_prefix)))
+        env.set('AOMP_GPU',
+                '`{0}/rocm-bin/mygpu`'.format(format(openmp_extras_prefix)))
+
+    def setup_build_environment(self, env):
+        openmp_extras_prefix = self.spec['rocm-openmp-extras'].prefix
+        llvm_prefix = self.spec['llvm-amdgpu'].prefix
+        env.set('AOMP', '{0}'.format(format(llvm_prefix)))
+        env.set('FC', '{0}/bin/flang'.format(format(openmp_extras_prefix)))
+        env.set(
+            'GFXLIST',
+            'gfx700 gfx701 gfx801 gfx803 gfx900 gfx902 gfx906 gfx908')
+
+    def patch(self):
+        src = self.stage.source_path
+        aomp_extras = '{0}/rocm-openmp-extras/aomp-extras/aomp-device-libs'
+        libomptarget = \
+            '{0}/rocm-openmp-extras/llvm-project/openmp/libomptarget'
+        flang = '{0}/rocm-openmp-extras/flang/'
+
+        filter_file(
+            '{ROCM_DIR}/amdgcn/bitcode', '{DEVICE_LIBS_DIR}',
+            aomp_extras.format(src) + '/aompextras/CMakeLists.txt',
+            aomp_extras.format(src) + '/libm/CMakeLists.txt',
+            libomptarget.format(src) + '/deviceRTLs/amdgcn/CMakeLists.txt')
+
+        # Openmp adjustments
+        filter_file(
+            '-nogpulib', '-nogpulib -nogpuinc',
+            libomptarget.format(src) + '/deviceRTLs/amdgcn/CMakeLists.txt')
+
+        filter_file(
+            '-x hip', '-x hip -nogpulib -nogpuinc',
+            libomptarget.format(src) + '/deviceRTLs/amdgcn/CMakeLists.txt')
+
+        filter_file(
+            '-c ', '-c -nogpulib -nogpuinc -I{LIMIT}',
+            libomptarget.format(src) + '/hostrpc/CMakeLists.txt')
+
+        filter_file(
+            r'\${ROCM_DIR}/hsa/include \${ROCM_DIR}/hsa/include/hsa',
+            '${HSA_INCLUDE}/hsa/include ${HSA_INCLUDE}/hsa/include/hsa',
+            libomptarget.format(src) + '/plugins/hsa/CMakeLists.txt')
+
+        filter_file(
+            '{ROCM_DIR}/hsa/lib', '{HSA_LIB}',
+            libomptarget.format(src) + '/plugins/hsa/CMakeLists.txt')
+
+        filter_file(
+            r'{ROCM_DIR}/lib\)',
+            '{HSAKMT_LIB})\nset(HSAKMT_LIB64 ${HSAKMT_LIB64})',
+            libomptarget.format(src) + '/plugins/hsa/CMakeLists.txt')
+
+        filter_file(
+            r'-L\${LIBOMPTARGET_DEP_LIBHSAKMT_LIBRARIES_DIRS}',
+            '-L${LIBOMPTARGET_DEP_LIBHSAKMT_LIBRARIES_DIRS} -L${HSAKMT_LIB64}',
+            libomptarget.format(src) + '/plugins/hsa/CMakeLists.txt')
+
+        filter_file(
+            r'-rpath,\${LIBOMPTARGET_DEP_LIBHSAKMT_LIBRARIES_DIRS}',
+            '-rpath,${LIBOMPTARGET_DEP_LIBHSAKMT_LIBRARIES_DIRS}' +
+            ',-rpath,${HSAKMT_LIB64}',
+            libomptarget.format(src) + '/plugins/hsa/CMakeLists.txt')
+
+        filter_file(
+            '{ROCM_DIR}/include', '{COMGR_INCLUDE}',
+            libomptarget.format(src) + '/plugins/hsa/CMakeLists.txt')
+
+        filter_file(
+            r'-L\${LLVM_LIBDIR}\${OPENMP_LIBDIR_SUFFIX}',
```

Comment:
If you use `string=True`, you won't need to escape special chars because it won't treat it as a regex.

Reply:
Noted. This can be addressed in the next version update.

Reply:
Decided to go ahead and push this change.